src/metadataGeneration/typeResolver.ts

```diff
@@ -813,7 +836,7 @@ export class TypeResolver {
       format: this.getNodeFormat(propertySignature),
       name: identifier.text,
       required,
-      type: new TypeResolver(propertySignature.type, this.current, propertySignature.type.parent, this.context).resolve(),
+      type: new TypeResolver(propertySignature.type, this.current, propertySignature.type.parent, this.context, propertySignature.type as ts.TypeReferenceType).resolve(),
```

Comment:
Maybe we can check if it's a ref. If it is, we can set is without casting, if it is not we don't need it afaik.